### PR TITLE
Tempo: Encode IDs as hexadecimal when downloading traces

### LINKED ISSUE #63003: [Tempo] Use consistent encoding when downloading traces as JSON
Traces downloaded via `Inspector -> Data -> Download traces` in trace view are downloaded in JSON format. This is a nice utility that allows to do further analysis over traces.

However, the encoding used for IDs (byte arrays) in Tempo, and almost everywhere is hexadecimal, while the default encoding JSON uses is base64. This causes traces downloaded this way to have a different encoding for IDs than everywhere else, including trace view itself and traces downloaded in CSV.

We should consider making the encoding consistent across all visualisations and formats.

### PR DESCRIPTION
**Pull request by kousikmitra**
**What is this feature?**

While downloading traces as json the ids are getting converted to base64. This change will remove the conversion and export ids in hex format instead. So that encoding remain consistent across.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #63003 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by adrapereira — APPROVED** [review]
LGTM! Thanks for the contribution @kousikmitra

**Review by adrapereira — CHANGES_REQUESTED** [review]
There's actually a couple of issues, sorry! 

Apart from my comment you also need to fix this test: 
```
FAIL public/app/plugins/datasource/tempo/datasource.test.ts (22.543 s)
  ● Tempo data source › should handle json file upload

    expect(received).toBe(expected) // Object.is equality

    Expected: "60ba2abb44f13eae"
    Received: "RPE+rg=="

      197 |     expect(field.name).toBe('traceID');
      198 |     expect(field.type).toBe(FieldType.string);
    > 199 |     expect(field.values.get(0)).toBe('60ba2abb44f13eae');
          |                                 ^
      200 |     expect(field.values.length).toBe(6);
      201 |   });
      202 |

      at public/app/plugins/datasource/tempo/datasource.test.ts:199:33
```

File: public/app/plugins/datasource/tempo/resultTransformer.ts [hunk 1]
```diff
@@ -297,9 +274,9 @@ export function transformFromOTLP(
       for (const librarySpan of data.instrumentationLibrarySpans) {
         for (const span of librarySpan.spans) {
           frame.add({
-            traceID: transformBase64IDToHexString(span.traceId),
-            spanID: transformBase64IDToHexString(span.spanId),
-            parentSpanID: transformBase64IDToHexString(span.parentSpanId || ''),
+            traceID: span.traceId.length > 16 ? span.traceId.slice(16) : span.traceID,
```
[the remark this answers is not in the document]
Comment (kousikmitra) [reply public/app/plugins/datasource/tempo/resultTransformer.ts:277]: my bad, fixed.

**Comment by kousikmitra** [comment]
Updated the mock response.

After this change the expectation for importing tempo trace from json is, the json data should have traceID, spanID and parentSpanId all in hex format instead of base64.

**Review by adrapereira — APPROVED** [review]
Thank you for the changes, approved!

**Pull request merged**